### PR TITLE
Add event metadata field to Alert spec

### PR DESCRIPTION
**Pull request by matheuscscp**
Fixes https://github.com/fluxcd/notification-controller/issues/511

### CODE REVIEW

File: internal/server/event_handlers.go [hunk 1]
```diff
@@ -256,13 +256,15 @@ func (s *EventServer) handleEvent() func(w http.ResponseWriter, r *http.Request)
...
-					notification.Metadata = map[string]string{
-						"summary": alert.Spec.Summary,
-					}
-				} else {
-					notification.Metadata["summary"] = alert.Spec.Summary
+				meta := notification.Metadata
+				if meta == nil {
+					meta = make(map[string]string)
+				}
+				for key, value := range alert.Spec.EventMetadata {
```
Comment (stefanprodan) [inline internal/server/event_handlers.go:263]: This should happen even if summary is string empty

Comment (aristapimenta) [reply internal/server/event_handlers.go:263]: I'm confused :thinking:

Comment (aristapimenta) [reply internal/server/event_handlers.go:263]: oh, right!!! lemme fix it

Comment (aristapimenta) [reply internal/server/event_handlers.go]: fixed :ok_hand:

Comment (stefanprodan) [reply internal/server/event_handlers.go]: Thanks, looks good. We'll need to wait for GitHub to recover as now Actions are down.

File: docs/spec/v1beta2/alerts.md [hunk 2]
```diff
@@ -162,6 +162,13 @@ starting the controller with the `--no-cross-namespace-refs=true` flag.
 When this flag is set, alerts can only refer to event sources in the same namespace as the alert object,
 preventing tenants from subscribing to another tenant's events.
 
+### Event metadata
+
+`.spec.eventMetadata` is an optional field for adding metadata to events emitted by the
+controller. Metadata fields added by the controller have priority over the fields
+added here, and the fields added here have priority over fields originally present
+in the event.
+
```
Comment (stefanprodan) [inline docs/spec/v1beta2/alerts.md:171]: Please add here an example section with metadata for:
```yaml
app.kubernetes.io/env : "production"
app.kubernetes.io/cluster: "my-cluster"
app.kubernetes.io/region: "us-east-1"
```

Comment (matheuscscp) [reply docs/spec/v1beta2/alerts.md:171]: done!

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @matheuscscp 🏅

File: api/v1beta2/alert_types.go [hunk 3]
```diff
@@ -50,6 +50,13 @@ type AlertSpec struct {
 	// +optional
 	InclusionList []string `json:"inclusionList,omitempty"`
 
+	// EventMetadata is an optional field for adding metadata to events emitted by the
+	// controller. Metadata fields added by the controller have priority over the fields
+	// added here, and the fields added here have priority over fields originally present
+	// in the event.
```
[the remark this answers is not in the document]
Comment (hiddeco) [reply api/v1beta2/alert_types.go:56]: After going through the PR, I have the same confusion.

[the remark this answers is not in the document]
Comment (matheuscscp) [reply api/v1beta2/alert_types.go:56]: the controller mentioned here which has priority over everything else is notification-controller. when the event arrives at notification-controller, first the metadata is cleaned up, then the `eventMetadata` field from the `Alert` spec is added, and then finally the `summary` field from the `Alert` spec is added (the only one at the moment). the priority is given by this order.

i will file a PR to improve this explanation

**Pull request merged**